### PR TITLE
Randomizes Cloning. Removes oxygen canister from Cloning.

### PR DESCRIPTION
**Pull request by FreeStylaLT**
This adds three separate spawners for each of the cloning's machinery equipment with varying chance of spawning.

The scanning and cloning pods have a 25% chance of spawning.
The console has 50% chance of spawning.

Examples from five rounds:
img
img
img
img
img

Obviously, not a good count to make big assumptions from, but as you can see, at least one object is now likely to be there, with a rare chance of either nothing or everything spawning.

This also removes the cryo oxygen canister. These changes are meant to
A) make death slightly more meaningful by removing the reliability of cloning _being_ there and to 
B) make more inter-departmental cooperation happen by removing the canister every time. Though, I am considering replacing it with a spawner as well, I'm not really sure about it yet.

🆑 FreeStylaLT
del: Removed Cloning's oxygen canister.
tweak: Cloning now has randomized setup each round.
/🆑

### CODE REVIEW

**Comment by StreakyHaddock** [comment]
i'd be willing to give it a try for a bit, with the caveat of a vote being held to see how much people hate/love it.

**Comment by granodd** [comment]
Better rev up that formaldehyde.

File: _maps/map_files/cyberiad/cyberiad.dmm [hunk 1]
```diff
@@ -5224,10 +5226,8 @@
...
-"bWC" = (/obj/machinery/computer/cloning,/turf/simulated/floor/plasteel{icon_state = "freezerfloor"},/area/medical/genetics_cloning)
-"bWD" = (/obj/machinery/dna_scannernew,/turf/simulated/floor/plasteel{icon_state = "freezerfloor"},/area/medical/genetics_cloning)
-"bWE" = (/obj/machinery/portable_atmospherics/canister/oxygen{name = "Canister: \[O2] (CRYO)"},/obj/structure/window/reinforced{dir = 8},/obj/machinery/door/window/eastright{dir = 1; icon_state = "left"; name = "Cryo Tank Storage"; req_access_txt = "0"; req_one_access_txt = "5;32"},/obj/machinery/atmospherics/unary/portables_connector{dir = 1},/turf/simulated/floor/plasteel{icon_state = "freezerfloor"},/area/medical/genetics_cloning)
```
Comment (Spartan6) [inline _maps/map_files/cyberiad/cyberiad.dmm]: Replace it with the yellow canister from maintenance, with nothing in it, for looks.

**Comment by FreeStylaLT** [comment]
seriously travis the fuck

**Comment by EvadableMoxie** [comment]
This feels more like roundstart homework that interdepartmental cooperation.

**Comment by Jountax** [comment]
Have to echo EvadableMoxie's thoughts. What is this going to achieve other than dead people who are found having to wait to be cloned?

If anything this will just make the CMO more likely to rush SR at round start and extend its use to everyone and then we might as well just remove cloning entirely and make geneticists only blong to the science department.

I mean, I wouldn't be entirely opposed to removing cloning entirely since there are alternatives to it anyway, but that would carry lore implications and IC implications for anyone who had cloning involved in their backstory somehow.

**Comment by Dinarzad** [comment]
Cloning is the baseline revival mechanic of SS13, it is almost a constant across every server/codebase I have ever played on.
Everything in the game is balanced around cloning being the "Standard" method of revival.
Cloning takes a chunk of time, can only do 1 person at a time and can be sabotaged/blown apart.

I **really** do not think you guys have fully thought out the massive game wide balance ramifications dicking with cloning this much is going to cause.
We can open Pandora's Box if you guys *reaaaaaaally* wanna, but I don't think you're going to like what comes out when you do it.

**Comment by taukausanake** [comment]
I don't understand the purpose of HOW this is being done. Either make all of cloning spawn or none of it, except don't because that's retarded. I do feel removing the oxygen canister isn't a problem; I don't feel departments should start out with everything to run but not starting with certain machinery? Let's just put all of R&D like this and call it a nerf

**Comment by SkeletalElite** [comment]
This has serious balance implications particularly in short rounds such as blob and nukie. I think wizard will be the most impacted as it lasts long enough for the effect to really take toll and short enough that not having at first can be really bad.

**Comment by Vivalas** [comment]
@Dinarzad

I feel that's exaggerated. I personally am of the opinion to remove it completely, because I feel death is way too cheap.

I hate having both defib and cloning, but having just defib by itself without cloning makes medical more interesting imo, since there is a reason not to just let people die, and to actually have to try to save critical patients.

Anyways, I dislike cryo as well for similar reasons but removing the oxygen canister just feels like a busywork thing. I mean, most "interdepartment" interaction things become just that. I feel it's best to just drop the idea of having medical rely on atmos and think of a different nerf to cryo.

**Comment by Jountax** [comment]
If you're trying to nerf cryo, does this even touch the cryocells outside of cloning?

**Comment by Dinarzad** [comment]
@Vivalis

'Ey, you guys can do whatever you want with Cloning, remove it, nerf it, buff it whatever you feel.
All I am saying is:
Cloning is a staple feature on almost very codebase, it is the primary balancing feature to everything else.
How easily antags can kill people exists because of how cloning can bring most people back so long as they are not husked and have a brain.  You remove cloning, you now have to adjust the power of Antagonists and... anything that can kill people, because once you past the point of defibrillation, you enter a realm where it is very possible to be offed fro mthe round for good, in relatively easy ways.

Certain species also exist in the state they exist in, because "Cannot be cloned" is one of their primary detriments, like Vox or Slime people (And kind of Plasmas).  You make Cloning half worthless and that becomes about as much of a detriment as Vox's "Must be wearing a mask" so THAT power balance is now skewed and also has to be revisited.

To say fiddling with cloning won't have widespread balance repercussions is to forget a lot of systems in this game and how they all interlink up.
You wanna make death hard, I'd sooner delete Defibs and SR, both are fast methods of revival with no real downside that can be done on several people at once in the right conditions.  Cloning can be sabotaged, takes a while and can only do one person at a time. It's a lot harder to go through a backlog of corpses with Cloning.  Death being trivial really only started to be a thing when those got added anyway.
Anyway, I've said my bit on it, I think screwed with this is going to involve having to rebalance half the game, how y'all take that or don't take that is up to you guys.

**Comment by Tayyyyyyy** [comment]
Bay doesn't have cloning. Baystation12/Baystation12#17126

So many comments on this PR are just "things are changing, oh no change bad". Cloning is too easy and "fire and forget". Lots of times it's easier to just let someone die and clone them. There needs to be a place somewhere in the middle where doctors are still incentivized to fix people, but also not have to spend forever fixing someone with bones broken in every part of their body. Whether this comes in the form of a heart/brain surgery that can make bodies defibable, or something else, is something that is still to be figured out.

Instead of always arguing for the status quo, consider proposing constructive criticism that actually moves the conversation forward instead of just shoots ideas down and engages in exaggerated apocalyptic rhetoric. This isn't even that horrible of a nerf. It only removes roundstart cloning. It's a step in the right direction and will encourage medical to do other things instead of cloning, as well as make death more impactful.

**Comment by Vivalas** [comment]
@Dinarzad

And the balance relationship you describe is something I consider as a fundamentally *stupid* design consideration in a game about paranoia and survival. I also feel your argument is about as opinionated as mine. Cloning in fact is an accident made into a feature after people abused a bug in pre-goon genetics that allowed reviving people. In my opinion the reliance you describe on it is unhealthy to the game. 

To put it in simple terms, yes, **dying with no chance of revival is something I very much like**. You can't have suspense, horror, drama, and genuine fear when all you have to do is scream "HELP" with your suit sensors on and you get another life 90% of the time.

The mere fact that people consider antags permakilling people as "rude" or "inconsiderate" reflects on the unhealthy meta come about from cloning being so prevalent in all codebases. It's a fucking *survival paranoia* game where they're the *bad guys*. If you die and sit the round out, well you learned a lesson and your actions had genuine weight. In most cases you can probably get re-involved in the round in one of the many ways added over time to keep ghosts happy.

**Comment by Dinarzad** [comment]
@Tayyyyyyy 

> Whether this comes in the form of a heart/brain surgery that can make bodies defibable, or something else, is something that is still to be figured out.

It's that last part that is the problem.
This is not the kind of change you can go "Just gotta do it an figure from there." You cannot just reach in an start ripping out key systems that have been balanced around for years, be they good or bad. You cause more harm then good that way and it ends up in a chaotic mess for ages until a solution IS finally put forward, and in the interim players have miserable experiences.
As for not being a big nerf
Science barely even builds themselves a telescience array, and sometimes can't even be assed into upgrading medical with a BRPED, what makes people think they'll be able to be assed into building cloning every round?

@Vivalas 

> I also feel your argument is about as opinionated as mine

I am not saying your argument IS opinionated. I don't know if yanking cloning out will hurt or benefit the game as a whole, every way I look at it, it comes up as a negative to me, but I am not sure.  What I *do* know though, is that you cannot half ass and cannot speed run it.   A system as engrained as Cloning you are talking about a fundamental system overhaul. It DOES effect balance in an EXTREMELY widespread fashion that you *have* to address **before** pulling it apart, otherwise the game has severe problems that players have to live with during the weeks or months that staff/maintainers/players all argue on what is the right way to proceed.

**Comment by uraniummeltdown** [comment]
This will just be tedious busywork that has to be done every round
I hope it doesn't get to the point where we consider removing Cloning, like Dinarzad said, the game is balanced around Cloning existing
It is way too easy and effortless to kill in this game. 

What's worse is that when you die on Paradise there is not much to do unless a midround event fires. /tg/ has ghost roles that are fun and while Bay has no Cloning they let you respawn. Paradise doesn't have Lavaland ghost roles or respawning enabled, and with a 2 hour round time that's painfully boring

**Comment by Tayyyyyyy** [comment]
Once again, all these arguments are off topic because they are based on the premise that this PR removes cloning. This PR does not do that.

**Comment by Rurik123** [comment]
The cheapness in death has been a problem for a while now, and for that, I more or less echo Tay's and Vivalas's arguments. I'd prefer it to be gone entirely and the defibs to be buffed, but hey, you gotta start somewhere.

**Comment by davidchan13** [comment]
If I had to pick between Cloning, Defibs and SR on which to keep, I'd take cloning everytime. I still recall the first time I died and was cloned in this game and thought it interesting that you grew in the pod, vaguely aware of the doctors around you until your body matured enough that your sight line might reach out across the room. Defibs are just spess magic so long as the corpse is recently fresh and mostly intact while SR being more of the same but not limited by battery life.

If someone wanted to rework how the cloner work, from something doctors would have to monitor and adjust settings to ensure the clone came out mostly normal, to being able to flash clone specific limbs and organs (including brains for immediate MMI), I'd be totally behind them. 

But leaving a partially set up cloning lab with no reliable source of parts to finish it on low pop rounds or hectic rounds like Wizard is bad. The only thing this does is add homework for responsible scientists to cough up cloner parts early in the round punishes medical staff anytime they get a random body dragged in from maint while science is in a noncooperative mood and doesn't care if the cloner hasn't been completed yet.

This PR is the equivalent of Brig cells having a random chance of being complete or just exceptionally wide hallways through Security with the Warden/HoS having to flag down Engineers every round to finish their brig. No engineers? Oops, sorry, can't properly detain criminals.

**Comment by taukausanake** [comment]
Here's a thought I might get behind. Make NT stingy bastards and have you order cloning and cryo tubes from Cargo. Maybe or maybe not come with basic components so you don't have to depend on Science to do their job. Also maybe up the R&D research for the boards if they are too low but that's another thing completely. I kind of agree that cloning is too cheap a solution for death so instead of removing or making it random chance of having just make it a little more expensive or work to get

**Comment by Vivalas** [comment]
@Dinarzad I know, what I was saying is that your argument is completely opinionated and/or generalized. Unless you are talking specifically about paradise, you can't say there is any sort of design direction that makes cloning a balance system. Beyond paradise, there is too much intermix between codebases for there really to be any over-arching balance / design meta.

Sure, people might take cloning into consideration while designing or balancing existing features, but claiming that removing it outright will horribly unbalance the game and ruin everything is, well, an opinion. And in my opinion, this opinion is baseless.

Anyways I thought of a crazy idea just now that might find common ground between the two stances on cloning and will start tossing it around on discord and the forums to see what people think.

**Comment by Aurorablade** [comment]
update meta also....

**Comment by Dumbdumn5** [comment]
The game is a little more balanced around cloning existing right now, with some races explicitly having downsides of not being able to be cloned. I'm cool with random placement, if only for variety's sake, but just outright having a chance for a major system of medbay to not exist is kind of... off in terms of how the round-to-round goes. I'd think it'd be cool if we could make cloning, defibbing, and other methods of revival have their own drawbacks and benefits. Could have cloning be more mutative, have somebody come out with club feet or more issues to fix, maybe a greater chance of negative mutations that would make a mutadone supply or competent geneticists a bit more necessary in crisis situations where chemistry might be gone/running low on chems. 

Still, outright or potential removal based on a random chance of one or more portions of cloning being missing is something that just feels off for things. Though the busywork argument may seem a bit weak in places, it has merit in that no other major system is often so offset, much like if we were to randomize R&D and potentially remove the autolathe or circuit printer and have them order parts in from cargo, as R&D isn't really always 100% needed. It's possible to run the station perfectly fine without any upgrades, though it does remove quite a bit of Sci's job and might be more tedious on everybody relying on them while they wait for sci to get things built.

Much like the prior example, many people rely on medical for cloning and revival, and though there are more rudimentary ways of reviving people, cloning is more of the 'refined' method that people rely on, especially in crisis situations like nukies, wizard, or higher population shadowling rounds. Made this after talking with the maints on this PR and I want to know if there's any thoughts in response to this.

**Comment by KasparoVy** [comment]
I retract my downvote, I could care less which way this goes :grin:. I would like to see the fallout. however.

**Comment by Dumbdumn5** [comment]
@Vivalas with things like antagonists and thinking about having to go through the effort of at least choosing the right corpse, the 'pop a new dude into the corpse' thing is a little off, but maybe genetic instability would be a good idea if done too much or if cloning somebody with organ damage might create imperfect or mutant organs. The random chance of it existing or not though feels just wrong, and kicking people out of their own bodies is really more of a punishment for the person just trying to play the game than a boon to the overall system.

**Comment by Vivalas** [comment]
I don't know what you mean by antagonists and choosing the right corpse, but in the case of punishing players by kicking them out of their body, yes, that is exactly the point. You died = you have a chance of being revived or it can go horribly wrong. Death needs to come with some punishment. I also like the idea of permanent consequences from death that you can't get rid of. I was thinking about brain damage and an ever increasing chance of continuity loss, but mutations and things more relevant than the current ones sound good too.

I honestly think death should be extremely punishing to those who suffer it, but most of my ideas are aimed as a sort of inbetween to those that think otherwise.

**Comment by Dinarzad** [comment]
If we really *have* to find some way of making life hell for new players and make Station Souls 13, make cloning cause Genetic deterioration. You can be cloned twice, after that, the DNA is too dangerously unstable to safely clone.
Effectively make a three strikes and your out system. If you're so careless as to be killed three times, then you can sit and think about it. It also encourages medical not to make cloning the go-to option, SR/Defibs used when they can be, so as not to burn through a players "Lives".

**Comment by taukausanake** [comment]
You know we did have #6590 that would have added punishment to getting cloned without a little extra work. It was closed in favor for the merged #6612 because #whynotboth was too extreme. If cloning is _still_ too powerful then maybe both wouldn't be so bad?

**Comment by davidchan13** [comment]
>Cloning too powerful.

My last 100 hours working in medbay, I'm willing to put antag tokens to bananas that 90% or more of corpses dragged into medbay were Defibbed or SR'd before anyone even tried using the cloning pod.

I need someone to explain to me what exactly they think is OP about cloning and why it shouldn't be in the game.

**Comment by Tayyyyyyy** [comment]
Can we please stop comparing cloning to defibbing and SR? They occupy different niches. Defibs require you to act really, really fast, and SR doesn't restore you back to 100% perfect condition. Obviously people are gonna try defibbing first, but most of the time, that body can't be brought back. Let's just talk about cloning instead of engaging in whataboutism.

**Comment by davidchan13** [comment]
@Tayyyyyyy They all occupy the same niche, reviving. Cloning is the least powerful, most time consuming of the three options. Medical gets 4+ Defibs and only require the body to be below a very generous damage threshold within ~5 minutes of death. SR can be made in the first 10~ minutes of a round if a chemist knows what they are doing. Instantly bringing a person back into the round. 

Cloning takes time, if not upgraded it does brain damage and can even inflict genetic defects onto a person. If it gets interrupted a person can be 'born' without limbs or even organs. AND you lose any prosthetics or implants you may have had when cloned.

If the goal of this is to get rid of prescanning, **_Then just code out prescanning or give it a T3+ requirement like autoprocess_** If there is some other justification for why cloning needs a nerf (when it's been nerfed repeatedly in the past while things like Defibs and SR were added) then I ask again; why do people seem to think the most complicated and time consuming method of revival is overpowered?

**Comment by Tayyyyyyy** [comment]
Ugh no they're so different. Defib is your first line of defense. You always try it before you either clone or SR (unless you can tell their body is rotting already). I mean, if you really wanna talk about this, defibs actually show how overpowered cloning is, cause sometimes the body is so messed up it's easier to just stick them in the cloner instead of defibbing them. Oh and defib is 3 minutes btw.

Cloning is overpowered because it's fire and forget to 100% health, skipping over medical care/surgery, and with autoprocessing, you even take out the fire part and it just turns into forget.

SR vs cloning is a more fair comparison, since both have the same time requirements. SR costs patches to use, and often requires surgery afterward. I don't know why you're so insistent that cloning is super shitty, that's clearly not true or people wouldn't use it all the time over everything else. Dying should have impact. It should not be a free ticket to 100% clean bill of health.

**Comment by davidchan13** [comment]
They are not near as different as you are trying to tout, your own description puts them in the same flow chart.

>Cloning is overpowered because it's fire and forget to 100% health, skipping over medical care/surgery, and with autoprocessing, you even take out the fire part and it just turns into forget.

This is flat out wrong. Cloning does genetic damage, and brain damage. And it takes a hell of a lot longer to clone someone than it defibbing. Slap synth patches, defib, send to surgery or cryo. Done. Autoprocess requires Tier 3 or better parts to even use, meaning this PR does **nothing** to address such a concern.

>that's clearly not true or people wouldn't use it all the time over everything else

By your own admission this isn't true, Defib is always used first, then SR is tried if there is any in the fridge. Cloning is next and Borging is used if cloning isn't an option for one reason or another.

Your own reasons prove my point. Cloning is the last resort, **_Everyone_** would rather use/be defibbed or have SR used before cloning iis attempted. I've never had a patient ask why I didn't clone them when defibbing or SRing them, but I've had dozens of people complain that I cloned them when defib or SR was an option.

**Comment by Tayyyyyyy** [comment]
> And it takes a hell of a lot longer to clone someone than it defibbing.

Please please please stop using this argument. After 3 minutes, it takes INFINITE time to revive someone with a defib. You're comparing apples and oranges!

Damage from cloning is solved by sticking them in the cryo cell conveniently provided right above the cloner.

Cloning is definitely not a last resort. If you've never seen doctors just let someone die so they could just clone them then you haven't played enough medbay.

**Comment by davidchan13** [comment]
Yeah okay, make a PR to remove Defibs and watch the backlash.

Cloning takes time. You can not get around this. Defibs are instant. That three minute window is the only balance to them. I've seen people defibbed in the bar, patched up and went back to drinking without ever going into medical.

And none of this argument goes into why removing round start cloning is in anyway a good idea.

**Comment by ManiacalSymphony** [comment]
Maybe it's time to roll out clone races

**Comment by Vivalas** [comment]
@Dinarzad Basically my idea, just with the added fun of random ghosts taking over your character ;)

**Comment by Jountax** [comment]
If people really want to nerf cloning somehow, you could always indirectly nerf it with a nerf to mutadone.

Seriously, it's insanely easy to make and a single unit will 100% of the time fix every single thing wrong with your genes.

**Comment by Vivalas** [comment]
But then we buff hulks..

**Comment by tigercat2000** [comment]
This is rather unnecessary - cloning has already experienced significant nerfs in the past, and it *really* isn't a better option compared to SR and Defibrillators. @Dumbdumn5 @DarkPyrolord

**Pull request closed without merging**